tests/utils/testapp/gen-syscall-events-callstack/gen-syscall-events-callstack.c

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright (C) - 2017 Francis Deslauriers <francis.deslauriers@efficios.com>
+ *
+ * This library is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by the
+ * Free Software Foundation; version 2.1 of the License.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public License
+ * for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this library; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301 USA
+ */
+
+#include <fcntl.h>
+#include <signal.h>
+#include <stdbool.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/syscall.h>
+#include <sys/types.h>
+#include <unistd.h>
+
+#include "utils.h"
+
+/**
+ * The process waits for the creation of a file passed as argument from an
+ * external processes to execute a syscall and exiting. This is useful for tests
+ * in combinaison with LTTng's PID tracker feature where we can trace the kernel
+ * events generated by our test process only.
+ */
+
+volatile int val = 0;
+
+long __attribute__ ((noinline))
+my_gettid(void)
+{
+    long ret;
+    asm volatile
+    (
+        "syscall"
+        : "=a" (ret)
+        : "0"(__NR_gettid)
+        : "cc", "rcx", "r11", "memory"
+    );
+    return ret;
```

Comment:
Can you add a x86 (32 bit) implementation of this syscall?
Also, we would need to disable this test if we're not building on x86/x64.

Reply:
I merged your syscall-generating test app a while ago. I'm guessing we can re-use the approach taken there instead of using an assembly stub?

Reply:
done.